stackpanel/src/main/java/org/vaadin/addons/stackpanel/client/StackPanelConnector.java

```diff
@@ -44,14 +47,16 @@ protected void extend(ServerConnector target) {
         DOM.insertChild(panel.captionNode, captionToggle, 2);
 
         updateStyleName(panel);
+       
 
         panel.addDomHandler(new ClickHandler() {
 
             @Override
             public void onClick(ClickEvent event) {
                 EventTarget eventTarget = event.getNativeEvent().getEventTarget();
-                if (eventTarget.cast() == panel.captionNode
-                        || eventTarget.cast() == panel.captionNode.getFirstChildElement()) {
+           
+                if (eventTarget.cast() == panel.captionNode || isChildofNode(panel, eventTarget)
+                		|| eventTarget.cast() == captionToggle.getChild(0))  { 
```

Comment:
Okay, lets remove the loop. I only like if you click not only on the "toggle button" but as well on caption or icon to open it. 
So finally the statement   ```eventTarget.cast() == panel.captionNode || eventTarget.cast() == captionToggle.getChild(0) || eventTarget.cast() == captionToggle.getFirstChildElement() ```
The middle statement is important that it toggles on the click on toggle icon